src/main/webapp/app/exercises/programming/manage/programming-exercise.component.ts

```diff
@@ -182,10 +182,19 @@ export class ProgrammingExerciseComponent extends ExerciseComponent implements O
     }
 
     openImportModal() {
-        const modalRef = this.modalService.open(ExerciseImportComponent, { size: 'lg', backdrop: 'static' });
+        const modalRef = this.modalService.open(ExerciseImportWrapperComponent, { size: 'lg', backdrop: 'static' });
         modalRef.componentInstance.exerciseType = ExerciseType.PROGRAMMING;
         modalRef.result.then((result: ProgrammingExercise) => {
-            this.router.navigate(['course-management', this.courseId, 'programming-exercises', 'import', result.id]);
+            //when the file is uploaded we set the id to undefined.
+            if (result.id === undefined) {
```

Comment:
Can we find a common place for this method, maybe in the import wrapper? This is now duplicated here and for the exams.